### PR TITLE
Configure dependabot

### PR DESCRIPTION
**Pull request by pellared**
## Why

In order to maintain and control the build dependencies

## What

- Update the dependabot config to bump GitHub Actions
- Remove unused GitHub workflows
-  Update the dependabot config to bump the Dockerfile base images used by CI build

## Tests

Testing in production 😄

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+      day: "friday"
+
+  # Maintain dependencies for Dockerfiles used in build pipeline
+  - package-ecosystem: "docker"
+    directory: "/tracer/build/_build/docker"
```
Comment (RassK) [inline .github/dependabot.yml:12]: There is a one more for tests: tests/test-applications/regressions/AutomapperTests, I guess tests/test-applications/ is good enough.

File: .github/workflows/verify_source_generated_changes_are_persisted.yml [hunk 2]
```diff
@@ -1,34 +0,0 @@
-name: Verify source generator changes have been persisted
```
Comment (Kielek) [inline .github/workflows/verify_source_generated_changes_are_persisted.yml:1]: Why? I think it is important to execute this task on PR/on main If it is working..

Comment (pellared) [reply .github/workflows/verify_source_generated_changes_are_persisted.yml:1]: You are right. I will revert it.

File: .github/dependabot.yml (hunk 1 above)
Comment (pellared) [reply .github/dependabot.yml:12]: I am not sure if we should bump these. They may be old versions as these may be regression tests for some concrete versions of .NET. I am more concerned about the "build" dependencies than the dependencies used just during tests. WDYT?

File: .github/workflows/verify_source_generated_changes_are_persisted.yml (hunk 2 above)
Comment (pellared) [reply .github/workflows/verify_source_generated_changes_are_persisted.yml:1]: Addressed 5451c3a1

File: .github/dependabot.yml (hunk 1 above)
Comment (pellared) [reply .github/dependabot.yml:12]: I see that there is no automation or real test for `AutomapperTests` (as well for most of the `tests/test-applications/regressions`

Comment (RassK) [reply .github/dependabot.yml:12]: I think you are right, that makes the bot spam about the versions that actually have to be different.

**Review by RassK — APPROVED** [review]
LGTM

**Review by Kielek — APPROVED** [review]

**Pull request merged**